File: src/types/video.ts
/**
 * TypeScript interfaces for video import
 * Matches Rust struct definitions for JSON serialization
 */

export interface VideoMetadata {
  duration: number;       // Duration in seconds
  resolution: string;     // e.g., "1920x1080"
  frame_rate: number;     // e.g., 29.97
  codec: string;          // e.g., "h264"
}

export interface ImportedClip {
  id: string;             // UUID v4
  filePath: string;       // Absolute file path
  duration: number;       // Duration in seconds
  thumbnail: string;      // Base64-encoded image (data:image/jpeg;base64,...)
  resolution: string;     // e.g., "1920x1080"
  frameRate: number;      // e.g., 29.97
  codec: string;          // e.g., "h264"
  importedAt: number;     // Timestamp in milliseconds
}

export interface ImportResult {
  success: boolean;
  clips: ImportedClip[];
  errors: string[];
}

export interface ImportState {
  isImporting: boolean;
  importQueue: string[];           // File paths waiting to import
  currentImportFile?: string;      // Current file being imported
  importProgress: number;          // 0-100 for current file
  lastImportDirectory?: string;    // Last directory user imported from
  error?: string;                  // Last error message
}

File: src-tauri/src/models/video.rs
use serde::{Deserialize, Serialize};

/// Video metadata extracted from file
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VideoMetadata {
    /// Duration in seconds
    pub duration: f64,
    /// Resolution (e.g., "1920x1080")
    pub resolution: String,
    /// Frame rate (e.g., 29.97)
    pub frame_rate: f64,
    /// Codec (e.g., "h264")
    pub codec: String,
}

/// Result of import operation
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportResult {
    /// Whether the import was successful
    pub success: bool,
    /// Successfully imported clips
    pub clips: Vec<ImportedClip>,
    /// Error messages for failed files
    pub errors: Vec<String>,
}

/// Extended clip data from import (includes thumbnail and metadata)
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportedClip {
    /// Unique identifier (UUID v4)
    pub id: String,
    /// Absolute file path
    pub filePath: String,
    /// Duration in seconds
    pub duration: f64,
    /// Base64-encoded thumbnail image
    pub thumbnail: String,
    /// Resolution (e.g., "1920x1080")
    pub resolution: String,
    /// Frame rate (e.g., 29.97)
    pub frameRate: f64,
    /// Codec (e.g., "h264")
    pub codec: String,
    /// Import timestamp (milliseconds since epoch)
    pub importedAt: i64,
}
